src/plugins/controls/public/controls/timeslider_control/components/time_slider_popover_content.tsx

```diff
@@ -27,29 +29,62 @@ interface Props {
   compressed: boolean;
 }
 
-export function TimeSliderPopoverContent(props: Props) {
-  const rangeInput = props.isAnchored ? (
+export function TimeSliderPopoverContent({
+  isAnchored,
+  setIsAnchored,
+  value,
+  onChange,
+  stepSize,
+  ticks,
+  timeRangeMin,
+  timeRangeMax,
+  compressed,
+}: Props) {
+  const [displayedValue, setDisplayedValue] = useState<Timeslice>(value ?? [0, 0]);
+
+  const debouncedOnChange = useMemo(
+    () =>
+      debounce((updateTimeslice: Timeslice | undefined) => {
+        onChange(updateTimeslice);
+      }, 350),
+    [onChange]
+  );
+  /**
+   * The following `useEffect` ensures that the changes to the value that come from the embeddable (for example,
+   * from the `clear` button on the dashboard) are reflected in the displayed value
+   */
+  useEffect(() => {
+    setDisplayedValue(value);
+  }, [value]);
+
+  const rangeInput = isAnchored ? (
     <TimeSliderAnchoredRange
-      value={props.value}
-      onChange={props.onChange}
-      stepSize={props.stepSize}
-      ticks={props.ticks}
-      timeRangeMin={props.timeRangeMin}
-      timeRangeMax={props.timeRangeMax}
-      compressed={props.compressed}
+      value={[displayedValue[0] || timeRangeMin, displayedValue[1] || timeRangeMax]}
+      onChange={(newValue) => {
+        setDisplayedValue(newValue as Timeslice);
+        debouncedOnChange(newValue);
+      }}
+      stepSize={stepSize}
+      ticks={ticks}
+      timeRangeMin={timeRangeMin}
+      timeRangeMax={timeRangeMax}
+      compressed={compressed}
     />
   ) : (
     <TimeSliderSlidingWindowRange
-      value={props.value}
-      onChange={props.onChange}
-      stepSize={props.stepSize}
-      ticks={props.ticks}
-      timeRangeMin={props.timeRangeMin}
-      timeRangeMax={props.timeRangeMax}
-      compressed={props.compressed}
+      value={[displayedValue[0] || timeRangeMin, displayedValue[1] || timeRangeMax]}
+      onChange={(newValue) => {
+        setDisplayedValue(newValue as Timeslice);
+        debouncedOnChange(newValue);
+      }}
+      stepSize={stepSize}
+      ticks={ticks}
+      timeRangeMin={timeRangeMin}
+      timeRangeMax={timeRangeMax}
+      compressed={compressed}
```

Comment:
We should definitely refactor this at some point - there is no reason to have these as two separate components 🤦 Not related to this PR, though - this is just super old code that hasn't been cleaned up in awhile.